shoop/admin/modules/orders/views/detail.py

```diff
@@ -99,10 +99,11 @@ def post(self, request, *args, **kwargs):
         old_status = order.status
         if new_status.role == OrderStatusRole.COMPLETE and not order.can_set_complete():
             raise Problem(_("Unable to set order as completed at this point"))
-        if new_status.role == OrderStatusRole.CANCELED and not order.can_set_canceled():
-            raise Problem(_("Paid, shipped, or canceled orders cannot be canceled"))
-        else:
-            order.set_canceled()
```

Comment:
This sets the status to cancelled anytime that new_status.role is not "canceled" (for example, if it's "complete").